src/main/java/org/sunbird/trending/controller/TrendingServiceImpl.java

```diff
@@ -178,84 +179,72 @@ private String getEndDateFormat() {
         return "";
     }
 
-    public Map<String, Object> trendingContentSearch(Map<String, Object> requestBody, String token) throws Exception {
+    public SBApiResponse trendingContentSearch(Map<String, Object> requestBody, String token) throws Exception {
+       SBApiResponse response = ProjectUtil.createDefaultResponse(Constants.MICROSITE_TOP_FEATURE_CONTENT_API);
+       try {
+           HashMap<String, Object> request = requestBody.containsKey(REQUEST) ? (HashMap<String, Object>) requestBody.get(REQUEST) : new HashMap<>();
+           if (request.isEmpty()) {
+               response.getParams().setStatus(Constants.FAILED);
+               response.put(MESSAGE, "Request is Missing");
+               response.setResponseCode(HttpStatus.BAD_REQUEST);
+               return response;
+           }
+           HashMap<String, Object> filter = request.get(FILTERS) == null ? new HashMap<>() : ((HashMap<String, Object>) request.get(FILTERS));
+           if (filter.isEmpty()) {
+               response.getParams().setStatus(Constants.FAILED);
+               response.put(MESSAGE, "Filter is Missing");
+               response.setResponseCode(HttpStatus.BAD_REQUEST);
+               return response;
+           }
+           List<String> contextTypeList = new ArrayList<>();
+           if (CollectionUtils.isNotEmpty(contextTypeList)) {
+               response.getParams().setStatus(Constants.FAILED);
+               response.put(MESSAGE, "ContextType is Missing");
+               response.setResponseCode(HttpStatus.BAD_REQUEST);
+               return response;
+           }
+           String org = (filter).get(ORGANISATION) == null ? "" : ((String) (filter).get(ORGANISATION));
+           if (org.isEmpty()) {
+               response.getParams().setStatus(Constants.FAILED);
+               response.put(MESSAGE, "Organization is Missing");
+               response.setResponseCode(HttpStatus.BAD_REQUEST);
+               return response;
+           }
+           Map<String, String> payloadToRedisKeyMapping = serverProperties.getPayloadToRedisKeyMapping();
+           contextTypeList.addAll(payloadToRedisKeyMapping.keySet());
```

Comment:
You have to read selected context Type from request payload not all contexts from properties.

Reply:
@SaipradeepR , so the response should be based only on "contextType" only ? In that case why we are having "contextType" values in properties file ?

Reply:
Implimented as suggested.